frontend/public/components/graphs/area.tsx

```diff
@@ -55,7 +62,21 @@ export const AreaChart: React.FC<AreaChartProps> = ({
   chartStatus,
 }) => {
   const [containerRef, width] = useRefWidth();
-  const getLabel = ({ datum: { x, y } }) => `${humanize(y).string} at ${formatDate(x)}`;
+  let processedData = data;
+  let unit: string = '';
+
+  if (isBinaryFunction(humanize)) {
+    let result: ProcessFrameResult;
+    if (humanize.toString() === humanizeBinaryBytes.toString()) {
```

Comment:
There is too much magic here, and this is likely to break. We should have a separate prop to opt in to this behavior rather than trying to determine what humanize function was passed.